overlord/snapstate/backend/setup.go

```diff
@@ -25,18 +25,26 @@ import (
 	"path/filepath"
 
 	"github.com/snapcore/snapd/boot"
+	"github.com/snapcore/snapd/osutil"
 	"github.com/snapcore/snapd/progress"
 	"github.com/snapcore/snapd/release"
 	"github.com/snapcore/snapd/snap"
 )
 
+// InstallUndoContext keeps information about what needs to be undone in case of install failure
+type InstallUndoContext struct {
+	// KeepTargetSnap indicates that the target .snap file under /var/lib/snapd/snap already existed when the
+	// backend attempted SetupSnap() through squashfs Install() and should be kept.
+	KeepTargetSnap bool `json:"keep-target-snap,omitempty"`
```

Comment:
and again, maybe then we turn around the POV here,  and call this  "TargetSnapExisted" ?

Reply:
I like that, thank you. It also makes it sound more like a "hint" for the undo which makes sense  given that I made it check if it's a symlink when deciding whether to keep it or remove.